packages/e2e-tests/specs/editor/various/copy-cut-paste-whole-blocks.test.js

```diff
@@ -92,4 +92,91 @@ describe( 'Copy/cut/paste of whole blocks', () => {
 		await pressKeyWithModifier( 'primary', 'v' );
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'should handle paste events once', async () => {
+		// Add group block with paragraph
+		await insertBlock( 'Group' );
+		await page.click( '.block-editor-button-block-appender' );
+		await page.click( '.editor-block-list-item-paragraph' );
+		await page.keyboard.type( 'P' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+		// Cut group
+		await pressKeyWithModifier( 'primary', 'x' );
+		await page.keyboard.press( 'Enter' );
+
+		await page.evaluate( () => {
+			window.e2eTestPasteOnce = [];
+			let oldBlocks = wp.data.select( 'core/block-editor' ).getBlocks();
+			wp.data.subscribe( () => {
+				const blocks = wp.data
+					.select( 'core/block-editor' )
+					.getBlocks();
+				if ( blocks !== oldBlocks ) {
+					window.e2eTestPasteOnce.push(
+						blocks.map( ( { clientId, name } ) => ( {
+							clientId,
+							name,
+						} ) )
+					);
+				}
+				oldBlocks = blocks;
+			} );
+		} );
+
+		// Paste
+		await pressKeyWithModifier( 'primary', 'v' );
+
+		// Blocks should only be modified once, not twice with new clientIds on a single paste action
+		const blocksUpdated = await page.evaluate(
+			() => window.e2eTestPasteOnce
+		);
+
+		expect( blocksUpdated.length ).toEqual( 1 );
```

Comment:
Interesting so, it looks like in the E2E test the clipboard `event.cancellable` is false, so a call to `event.preventDefault()` won't actually modify `event.defaultPrevented`

<img width="1270" alt="CleanShot 2021-09-16 at 13 59 49@2x" src="https://user-images.githubusercontent.com/1270189/133684995-3bc16892-8b78-4df7-b7b3-ccf687513d9d.png">